rules/DowngradePhp74/Rector/MethodCall/DowngradeReflectionGetTypeRector.php

```diff
@@ -62,42 +63,53 @@ public function run(ReflectionProperty $reflectionProperty)
      */
     public function getNodeTypes(): array
     {
-        return [MethodCall::class];
+        return [MethodCall::class, Instanceof_::class, FuncCall::class];
     }
 
     /**
-     * @param MethodCall $node
+     * @param MethodCall|Instanceof_|FuncCall $node
      */
-    public function refactor(Node $node): ?Node
+    public function refactor(Node $node): Node|null|int
     {
-        if (! $this->isName($node->name, 'getType')) {
-            return null;
+        if ($node instanceof Instanceof_) {
+            return $this->checkInstanceof($node);
         }
 
-        if (! $this->isObjectType($node->var, new ObjectType('ReflectionProperty'))) {
+        if ($node instanceof FuncCall) {
+            return $this->checkFuncCall($node);
+        }
+
+        if (! $this->isName($node->name, 'getType')) {
             return null;
         }
 
-        $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
-        if ($parentNode instanceof Instanceof_) {
+        if (! $this->isObjectType($node->var, new ObjectType('ReflectionProperty'))) {
             return null;
         }
 
         $args = [new Arg($node->var), new Arg(new String_('getType'))];
 
-        $ternary = new Ternary(
+        return new Ternary(
             $this->nodeFactory->createFuncCall('method_exists', $args),
             $node,
             $this->nodeFactory->createNull()
         );
+    }
 
-        $parentNode = $node->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parentNode instanceof Ternary) {
-            return $ternary;
+    private function checkInstanceof(Instanceof_ $instanceof): ?int
+    {
+        // skip instance of on call
+        if ($instanceof->expr instanceof MethodCall) {
+            return NodeTraverser::STOP_TRAVERSAL;
```

Comment:
@TomasVotruba this stop traversal somehow the culprit, it cause next rule just stopped loop back. It seems traverse is from `beforeTraverse()` which on rector rule, it actually root stmt instead of parent of target node, so it stop all process.

Reply:
@TomasVotruba it seems can be fixed with updated to use `return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;` I will update.

Reply:
> @TomasVotruba it seems can be fixed with updated to use return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN; I will update.

You're right, I've used the wrong constant. Thanks for the fix :+1:

Reply:
Using `NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN` seems cause error in test, I will look a possible fix

Reply:
`RectorNodeTraverser` seems need to be refactored to always reset `stopTraversal` value to false after refactor, see PR:

- https://github.com/rectorphp/rector-src/pull/4182

Reply:
visit per rector rule seems cause different issue, I am reverting at:

- https://github.com/rectorphp/rector-src/pull/4183

I will remove the STOP_TRAVERSAL, custom attribute, eg: `is_from_ternary` is probably the way to go.